applications/DEMApplication/custom_utilities/inlet.cpp

```diff
@@ -41,7 +41,7 @@ namespace Kratos {
 
     /// Constructor
 
-    DEM_Inlet::DEM_Inlet(ModelPart& inlet_modelpart): mInletModelPart(inlet_modelpart)
+    DEM_Inlet::DEM_Inlet(ModelPart& inlet_modelpart, int& seed, bool& use_external_seed): mInletModelPart(inlet_modelpart)
```

Comment:
Instead of two arguments, we could use just one, with a default value. Did you discard that option for some reason?

Reply:
I think it is better to use two to have more control when using an external seed

Reply:
What I mean is that you can put the seed argument to a default argument. That means that you can provide ir or not. If you don't provide that argument, a default value is assigned, which is exactly what you are doing, no? 
https://en.cppreference.com/w/cpp/language/default_arguments 
You don't lose control by using this feature, as far as I can see.

Reply:
But I do not want to provide a default argunment if I do not set an external seed but a random number.

Reply:
Is there any difference between a random number which is always the same and a fixed number?

Reply:
When an external seed is not used, the seed is set as rd() in inlet.cpp. It is not the default value assigned in `dem_default_parameters.py`

Reply:
If the random number is not the same we have a problem. We will not be able to debug cases, because it will behave differently every time we run a case.

Reply:
To debug a case you can use a fixed seed. Our goal in this implementation is to be able to run cases with the same distribution of particles in the inlet and run cases varying that distribution. Then, for the latter, it is needed random numbers.

Reply:
I just had a call with @farrufat-cimne about this. We think that the flag is not necessary. Just the seed will do. 
An uncontrolled rd() in the middle will give problems in the future, IMO, especially if we decide to keep a version of the constructor which is backward compatible (the seed has a default value).

Reply:
Ok